packages/block-library/src/gallery/edit.js

```diff
@@ -96,6 +93,69 @@ function GalleryEdit( props ) {
 		blockEditorStore
 	);
 
+	const {
+		imageSizes,
+		mediaUpload,
+		getMedia,
+		wasBlockJustInserted,
+	} = useSelect( ( select ) => {
+		const settings = select( blockEditorStore ).getSettings();
+		const lastBlockInserted = select(
+			blockEditorStore
+		).getLastBlockInserted();
+
+		return {
+			imageSizes: settings.imageSizes,
+			mediaUpload: settings.mediaUpload,
+			getMedia: select( coreStore ).getMedia,
+			wasBlockJustInserted:
+				!! lastBlockInserted &&
+				lastBlockInserted.clientId === clientId &&
+				lastBlockInserted.source === 'inserter_menu',
+		};
+	} );
+
+	const { resizedImages } = useMemo( () => {
```

Comment:
The dependencies array of `useSelect` is used to memoize the `mapSelect`, so the same `mapSelect` is invoked on every state change unless the dependencies change. For this case, this doesn't help us to improve the performance because this code is still being executed on every store change, so I had to move this logic inside a `useMemo` hook.

Reply:
Good eye 😉 I am seeing the correlation to the diff on the previous PR that utilized `useSelect` when computing the `resizedImages`.